dbloader/riak_loader.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import gevent
+from gevent.pool import Pool
+import json
+import random
+import riak as r
+import time
+from . import Loader
+from . import logger
+
+
+class RiakLoader(Loader):
+    ''' Class for load testing Riak. '''
+
+    def __init__(self):
+        '''
+        Initialize a RiakLoader
+        '''
+        Loader.__init__(self)
+        self.dbtype = 'Riak'
+        self.databases = ['rb_1', 'rb_2', 'rb_3']
+        self.db_prefix = 'riak_load_'
+        self.conn = None
+
+    def get_connection(self):
+        '''
+        Get a riak connection
+        '''
+        try:
+            self.conn = r.RiakClient(protocol=self.protocol,
+                                     host=self.host,
+                                     port=self.port)
+
+        except Exception:
+            logger.exception('Unable to connect to Riak')
+            return False
+        return
+
+    def insert(self, bucket, table=None, custom=None):
+        '''
+        Insert a single key/value
+        '''
+
+        logger.debug(' - Insert key %d', custom)
+        start_time = time.time()
+        if not self.conn:
+            self.get_connection()
+        try:
+            b = self.conn.bucket(bucket)
+            random_text = Loader.big_string(100)
+            value = {"type": "Load Test",
+                     "randString": random_text,
+                     "created": start_time,
+                     "custom": custom,
+                     "concurrency": 1}
+            kv = b.new(key=str(custom), data=value, content_type='application/json')
+            result = kv.store()
+            stored = b.get(kv.key)
+
+        except Exception:
+            logger.exception('Unable to insert an object')
+            return False
+        logger.debug(' - Inserted %s (%s)', result.key, kv.data)
+        logger.debug(' - Saved! (%s)', stored.data)
+        return time.time() - start_time
+
+    def update(self, bucket, table=None, custom=None):
+        '''
+        Update a single key/value
+        '''
+
+        start_time = time.time()
+        if not self.conn:
+            self.get_connection()
+        try:
+            b = self.conn.bucket(bucket)
+            kv = b.get(str(custom))
+            random_text = Loader.big_string(100)
+            value = {"type": "Load Test",
+                     "randString": random_text,
+                     "created": start_time,
```

Comment:
We shouldn't reset the created time here as we are just updating.

Reply:
Correct, in-fact, we should only need to update the `randString` and `updated` fields - thanks